spec/database/postgresql_spec.rb

```diff
@@ -239,6 +261,35 @@ module Backup
         end
       end # describe '#sudo_option'
 
+      describe "#pgdump_arguments" do
+        context "when url is not set" do
+          let(:option_methods) do
+            %w(
+              username_option connectivity_options user_options tables_to_dump
+              tables_to_skip name
+            )
+          end
+
+          it "returns username and connectivity options as separate arguments" do
+            option_methods.each { |name| db.stubs(name).returns(name) }
+            expect(db.send(:pgdump_arguments)).to eq option_methods.join(" ")
+          end
+        end
+
+        context "when url is set" do
+          before do
+            db.url = "postgres://username:password@localhost:5432/db_name"
+          end
+
+          let(:option_methods) { %w(user_options tables_to_dump tables_to_skip) }
+
+          it "returns an url with username and connectivity options included" do
+            option_methods.each { |name| db.stubs(name).returns(name) }
+            expect(db.send(:pgdump_arguments)).to eq "#{option_methods.join(" ")} #{db.url}"
+          end
+        end
+      end
```

Comment:
I don't like how it's testing a private method and stubbing all the option methods. But since that's how the rest of this spec is set up I'll allow it.